src/lib/utils/balancer/helpers/sor/sorManager.ts

```diff
@@ -171,6 +176,16 @@ export class SorManager {
     };
   }
 
+  getSwapAmount(amountScaled: OldBigNumber): BigNumber | void {
+    try {
+      return BigNumber.from(amountScaled.toString());
+    } catch (e) {
+      // BigNumber.from() throws an error when there is more than 15 digits
+      // not sure that we should process this specifically
+      console.log(e);
```

Comment:
Again I don't think this is a good idea, we should let it throw the error so we can handle that. Instead, we should be preventing the user from inputting more than 15 digits.

Reply:
yeah, input limit seems better here

Reply:
There should already be an input limit on the TokenInput so could you check why that isn't working in this case? Maybe it's a copy and past issue or something like that.